test/fakeintake/aggregator/common.go

```diff
@@ -23,8 +24,9 @@ type PayloadItem interface {
 type parseFunc[P PayloadItem] func(payload api.Payload) (items []P, err error)
 
 type Aggregator[P PayloadItem] struct {
-	payloadsByName map[string][]P
-	parse          parseFunc[P]
+	payloadsByName    map[string][]P
+	collectedAtByName map[string][]time.Time
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
What about adding `GetCollectedTime` to `PayloadItem` interface ? This implementation makes it hard to know at which moment one specific payload was received

Reply:
Thanks, changing the interface was the right way